agent/src/runtime_connectors/runtime_facade.rs

```diff
@@ -155,39 +155,37 @@ impl<
     ) -> (JoinHandle<()>, Workload) {
         let runtime = self.runtime.to_owned();
         let update_state_tx = update_state_tx.clone();
+        let workload_name = workload_spec.instance_name.workload_name().to_owned();
 
         let (control_interface_path, control_interface) = if let Some(info) = control_interface_info
         {
             let run_folder = info.get_run_folder().clone();
             let output_pipe_sender = info.get_to_server_sender();
             let instance_name = info.get_instance_name().clone();
             let authorizer = info.move_authorizer();
-            let control_interface = match ControlInterface::new(
-                &run_folder,
-                &instance_name,
-                output_pipe_sender,
-                authorizer,
-            ) {
-                Ok(control_interface) => Some(control_interface),
+            match ControlInterface::new(&run_folder, &instance_name, output_pipe_sender, authorizer)
```

Comment:
This would be the only solution to have similar signatures to the rest of functions (that need a ControlInterfaceInfo object) and I also discussed with @koppor about moving stuff around and using the ControlInterfaceInfo.

Reply:
I understand.
We can make the improvements at some other point when we have some spare time: #377